aspnet/migration/http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs

```diff
@@ -0,0 +1,316 @@
+            string contentMainType = mediaHeaderValue?.Type;    // ex. application
+            string contentSubType = mediaHeaderValue?.SubType;  // ex. x-www-form-urlencoded
+
+            System.Text.Encoding requestEncoding = mediaHeaderValue?.Encoding;
+            //<14
+
+
+            // ----------
+
+            // ASP.NET 4 - HttpContext.Request.Form 
+            //>15
+            if (httpContext.Request.HasFormContentType)
+            {
+                IFormCollection form;
+
+                form = httpContext.Request.Form; // sync
+                // Or
+                form = await httpContext.Request.ReadFormAsync(); // async
+
+                string firstName = form["firstname"];
+                string lastName = form["lastname"];
+            }
+            //<15
+            // ----------
+            // See 
+            // http://stackoverflow.com/questions/31389781/read-body-twice-in-asp-net-5/
+
+            // ASP.NET 4 - HttpContext.Request.InputStream
+            // Unlike reading the form, reading the raw body is not from a buffer.
+            // So you can only do this once per request.
+            // 
+            // If you need to read the body multiple times per request, see
+            // http://stackoverflow.com/questions/31389781/read-body-twice-in-asp-net-5/
+
+            //>16
+            string inputBody;
+            using (var reader = new System.IO.StreamReader(
+                httpContext.Request.Body, System.Text.Encoding.UTF8))
+            {
+                inputBody = reader.ReadToEnd();
+            }
+            //<16
+
+            // Use this middleware as a handler, so no call to 
+            // await _next.Invoke(aspnet5HttpContext);
+
+            // ==================================================================
+            // Response
+
+            // ASP.NET 4 - HttpContext.Response.Status  
+            // ASP.NET 4 - HttpContext.Response.StatusDescription (obsolete, removed from HTTP/2) 
+            //>30
+            // using Microsoft.AspNet.Http;
+            httpContext.Response.StatusCode = StatusCodes.Status200OK;
+            //<30
+
+            // ASP.NET 4 - HttpContext.Response.ContentEncoding and HttpContext.Response.ContentType  
+            //>31
+            // using Microsoft.Net.Http.Headers;
+            var mediaType = new MediaTypeHeaderValue("application/json");
+            mediaType.Encoding = System.Text.Encoding.UTF8;
+            httpContext.Response.ContentType = mediaType.ToString();
+            //<31
+
+            // ASP.NET 4 - HttpContext.Response.ContentType only 
+            //>32
+            httpContext.Response.ContentType = "text/html";
+            //<32
+
+            // ----------------------
+            // ASP.NET 4 - HttpContext.Response.Output  
+            //
+            // If you want to serve up the contents of a file, see
```

Comment:
What does serving file content have to do with Response.Output? Wouldn't TransmitFile be more relevant?